idea_town/experiments/models.py

```diff
@@ -60,6 +61,18 @@ class UserInstallation(models.Model):
     created = models.DateTimeField(auto_now_add=True)
     modified = models.DateTimeField(auto_now=True)
 
+    def save(self, *args, **kwargs):
+        if self.pk is None:
+            self.experiment.installation_count = models.F('installation_count') + 1
+            self.experiment.save()
+        super(UserInstallation, self).save(*args, **kwargs)
+
+    def delete(self, *args, **kwargs):
+        if self.experiment.installation_count > 0:
+            self.experiment.installation_count = models.F('installation_count') - 1
```

Comment:
Multithreading is fun: what's true when line 71 executes may no longer be true when line 72 executes. Without an atomic read-then-write statement, you can't be sure what the value is when line 72 executes.

To ensure you don't clobber any other writes, you should get rid of that positive value check on line 71.

Enforcing the counter > 0 thing is a challenge, and I'm not sure of the best way to handle it. To keep things moving, I'd say you should be fine to remove line 71, file a followup bug to handle the counter < 0 case, and see how @lmorchard wants to handle it when he's back.